extensions/src/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoService.java

```diff
@@ -17,15 +17,149 @@
 
 package org.pentaho.platform.web.http.api.resources.services;
 
+import org.pentaho.platform.api.engine.IAuthorizationPolicy;
+import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.api.engine.security.userroledao.IPentahoRole;
+import org.pentaho.platform.api.engine.security.userroledao.IPentahoUser;
 import org.pentaho.platform.api.engine.security.userroledao.IUserRoleDao;
+import org.pentaho.platform.api.engine.security.userroledao.NotFoundException;
+import org.pentaho.platform.api.engine.security.userroledao.UncategorizedUserRoleDaoException;
+import org.pentaho.platform.api.mt.ITenant;
+import org.pentaho.platform.core.mt.Tenant;
 import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
 import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.pentaho.platform.security.policy.rolebased.actions.AdministerSecurityAction;
+import org.pentaho.platform.security.policy.rolebased.actions.RepositoryCreateAction;
+import org.pentaho.platform.security.policy.rolebased.actions.RepositoryReadAction;
+import org.pentaho.platform.web.http.api.resources.RoleListWrapper;
 import org.pentaho.platform.web.http.api.resources.UserListWrapper;
 
+import java.util.HashSet;
+import java.util.Set;
+import java.util.StringTokenizer;
+
 public class UserRoleDaoService {
-    public UserListWrapper getUsers() throws Exception {
-        IUserRoleDao roleDao =
-                PentahoSystem.get( IUserRoleDao.class, "userRoleDaoProxy", PentahoSessionHolder.getSession() );
-        return new UserListWrapper( roleDao.getUsers() );
+  public UserListWrapper getUsers() throws Exception {
+    IUserRoleDao roleDao =
+      PentahoSystem.get( IUserRoleDao.class, "userRoleDaoProxy", PentahoSessionHolder.getSession() );
```

Comment:
Replace this with simply PentahoSystem.get( IUserRoleDao.class );

Also cache the userRoleDao as a member variable of this class so you're not looking it up every time.